### ISSUE TITLE
Vending machines will discount rotten goods

### ISSUE THREAD

**Issue by chrispikula** [CleverRaven/Cataclysm-DDA#47855]
# Describe the bug

Vending machines put the price of what they are selling to what it currently is, rather than what it was stocked with originally.

# Steps To Reproduce

Look at a vending machine with food, advance time till some of it rots, look at it again.  Also, you can buy free plastic bags once the items are completely gone.

# Expected behavior

That the vending machines don't act is if they have an AI that can determine new item prices for goods gone bad.  

# Versions and configuration

0.E-9867-gadceb7b , Build 11501

# Additional context

Probably do to the same logic as #38379

**Comment by actual-nh** [comment]
Still happening?

**Comment by chrispikula** [comment]
I haven't had a play through where I found an intact vending machine with rotable goods in a while, so I'm not sure.

**Comment by actual-nh** [comment]
@Night-Pryanik, have you seen this? I know you recently did a PR involving vending machines.

**Comment by Night-Pryanik** [comment]
Nope, I didn't activate the machines when I was working on them, I only visually checked their state (intact/broken).
